app/scripts/components/form/dropdown.js

```diff
@@ -21,39 +23,68 @@ function renderMenu (items, value, style) {
   );
 }
 
+/**
+ * Translator for status objects.
+ *
+ * @param {Object} statusObject
+ * @param {string} status - url query parameter to match. e.g. "running", "completed", "failed"
+ * @returns {string} - either the correct label from the status Object, or the input status.
+ */
+const statusToLabel = (statusObject = {}, status) => {
+  let result = Object.keys(statusObject).filter(
+    (label) => statusObject[label] === status
+  );
+  if (result.length) return result[0];
+  return status;
+};
+
 class Dropdown extends React.Component {
-  constructor () {
-    super();
+  constructor (props) {
+    super(props);
     this.displayName = 'Dropdown';
+    const initialValue = statusToLabel(props.options, initialValueFromLocation(props));
     this.state = {
-      value: ''
+      value: initialValue
     };
     this.onSelect = this.onSelect.bind(this);
     this.onChange = this.onChange.bind(this);
   }
 
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    const {location, paramKey, dispatch, action, options} = this.props;
+    if (location.query[paramKey] !== prevProps.location.query[paramKey]) {
```

Comment:
Hmm, is `location.query` always defined, even on initial component load? I'm surprised that this never errors.

Reply:
By creating the router history with `useRouterHistory(createHashHistory)()`, there will always be a query object on a location.   It may be empty, but then these are undefined and not errors.